app/tests/components_tests/test_forms.py

```diff
@@ -24,3 +44,145 @@ def test_interface_form_field_image_queryset_filter():
     assert im2 not in field.field.fields[0].queryset.all()
     assert upload1 in field.field.fields[1].queryset.all()
     assert upload2 not in field.field.fields[1].queryset.all()
+
+
+@pytest.mark.parametrize(
+    "form_class,base_object_factory",
+    (
+        (
+            DisplaySetCreateForm,
+            ReaderStudyFactory,
+        ),
+        (
+            ArchiveItemCreateForm,
+            ArchiveFactory,
+        ),
+    ),
+)
+@pytest.mark.django_db
+def test_image_widget_current_value_in_archive_item_and_display_set_create_forms(
+    form_class, base_object_factory
+):
+    user = UserFactory()
+    image_ci = ComponentInterfaceFactory(kind=ComponentInterface.Kind.IMAGE)
+
+    image = ImageFactory()
+    assign_perm("cases.view_image", user, image)
+
+    user_upload = UserUploadFactory(creator=user)
+    user_upload.status = UserUpload.StatusChoices.COMPLETED
+    user_upload.save()
+
+    form1 = form_class(
+        user=user,
+        instance=None,
+        base_obj=base_object_factory(),
+        data={
+            "order": 1,

```

Comment:
order is only necessary for display sets. It will just be ignored on archive item forms, but it's cleaner to add another variable to parametrize `extra_form_kwargs` and to pass that to data `**extra_form_kwargs`.

Reply:
Done